go/pkg/filemetadata/filemetadata.go

```diff
@@ -112,10 +114,12 @@ func Compute(filename string) *Metadata {
 		}
 		xattrValue, err := XattrAccess.getXAttr(filename, XattrDigestName)
 		if err == nil {
-			md.Digest = digest.Digest{
-				Hash: string(xattrValue),
-				Size: file.Size(),
+			xattrStr := string(xattrValue)
+			if strings.Contains(xattrStr, "/") {
+				md.Digest, md.Err = digest.NewFromString(xattrStr)
```

Comment:
Can you make sure md.Err is a FileError{} similar to other code paths above?

Reply:
I have changed all the errors below as FileError{}
1. compute from hash/size xattr
2. compute from hash_only xattr
3. compute directly from file without xattr